pruner/full/pruner.go

```diff
@@ -4,36 +4,29 @@ import (
 	"context"
 	"errors"
 
-	"github.com/filecoin-project/dagstore"
 	logging "github.com/ipfs/go-log/v2"
 
 	"github.com/celestiaorg/celestia-node/header"
-	"github.com/celestiaorg/celestia-node/share"
-	"github.com/celestiaorg/celestia-node/share/eds"
+	"github.com/celestiaorg/celestia-node/store"
 )
 
 var log = logging.Logger("pruner/full")
 
 type Pruner struct {
-	store *eds.Store
+	store *store.Store
 }
 
-func NewPruner(store *eds.Store) *Pruner {
+func NewPruner(store *store.Store) *Pruner {
 	return &Pruner{
 		store: store,
 	}
 }
 
 func (p *Pruner) Prune(ctx context.Context, eh *header.ExtendedHeader) error {
-	// short circuit on empty eds
-	if eh.DAH.Equals(share.EmptyEDSRoots()) {
-		return nil
-	}
-
 	log.Debugf("pruning header %s", eh.DAH.Hash())
 
-	err := p.store.Remove(ctx, eh.DAH.Hash())
-	if err != nil && !errors.Is(err, dagstore.ErrShardUnknown) {
+	err := p.store.Remove(ctx, eh.Height(), eh.DAH.Hash())
+	if err != nil && !errors.Is(err, store.ErrNotFound) {
```

Comment:
Store doesn't return this error. The method is now idempotent.
```suggestion
	err := p.store.Remove(ctx, eh.Height(), eh.DAH.Hash())
	if err != nil {
```